### ISSUE TITLE
Allow alternative fs to be used for reading files.

### ISSUE THREAD

**Issue by dazinator** [dependents/node-filing-cabinet#65]
Driver for this is: pahen/madge#186
I'll submit a PR

It looks like the only place `fs` is used currently is to read a tsconfig file here: https://github.com/dependents/node-filing-cabinet/blob/master/index.js#L197

**Comment by dazinator** [comment]
Actually I'll probably have to get the typescript compiler / host to read files from the alternate fs implementation too. There are hooks for that I beleive.

**Comment by XhmikosR** [comment]
I have prepared a branch, but I'm not sure if this is still needed or if it makes sense anymore: https://github.com/dependents/node-filing-cabinet/compare/main...xmr/alt-fs

**Comment by dazinator** [comment]
@XhmikosR Thanks - I think this is a helpful change that increases the flexibility of the library to unblock future work sitting on top potentially with Madge. If you are comfortable with it - It would be good to see it merged - and then one day soon I might pick up the work in madge again that leverages this. Thanks

**Comment by XhmikosR** [comment]
@dazinator: I created #164, please give it a go or a review and let me know.
